tools/fuse-waked/fuse-waked.cpp

```diff
@@ -697,6 +697,19 @@ static int wakefuse_unlink_trace(const char *path)
 	return out;
 }
 
+// Check if the directory has children that were written and not yet deleted.
+// Note: we don't check the 'files_visible' list as when the directory itself or
+// a file within that directory is visible then is_writable() would have already failed.
+static bool has_written_children(const std::string& dir, Job& job) {
+	for (auto &written : job.files_wrote) {
+		if (written.size() > dir.size() && written[dir.size()] == '/' && written.find(dir) == 0) {
```

Comment:
`find()` could be slow because it'll attempt to continue searching past the 0th index, even though that you only care about whether `dir` is a proper prefix of `written`. It's too bad that only C++20 added [`starts_with()`](https://en.cppreference.com/w/cpp/string/basic_string/starts_with), but I think you can at least use [`compare()`](https://cplusplus.com/reference/string/string/compare/) and just specify the start position and length to compare.

Reply:
Not to mention the outer loop is unnecessary. Try `lower_bound(dir + "/")` followed by check vs. `end` and a `compare`.